slack_sdk/socket_mode/aiohttp/__init__.py

```diff
@@ -200,19 +256,25 @@ async def connect(self):
             self.wss_uri = await self.issue_new_wss_url()
         self.current_session = await self.aiohttp_client_session.ws_connect(
             self.wss_uri,
+            autoping=False,
             heartbeat=self.ping_interval,
             proxy=self.proxy,
         )
         self.auto_reconnect_enabled = self.default_auto_reconnect_enabled
+        self.stale = False
         self.logger.info("A new session has been established")
 
-        if self.current_session_monitor is None:
-            self.current_session_monitor = asyncio.ensure_future(
-                self.monitor_current_session()
-            )
+        if self.current_session_monitor is not None:
+            self.current_session_monitor.cancel()
```

Comment:
Also, while doing more tests (=aggressively & repeatedly doing force reconnection), I found that the Task object may refer to old abandoned objects. Thus, cancelling the old Task this way makes this code much safer.

Reply:
Some part of the session object may be related to the behavior but this part is our own code. Thus, no need to file an issue to the project. Basically, we've improved our code this time to be safer 😅